modules/tinymce/src/core/main/ts/delete/TableDelete.ts

```diff
@@ -66,8 +70,24 @@ const setCursorInCell = (editor: Editor, cell: SugarElement<HTMLTableCellElement
  * - part of a table and content outside is selected
  */
 const emptySingleTableCells = (editor: Editor, cells: SugarElement<HTMLTableCellElement>[], outsideDetails: Optional<OutsideTableDetails>): boolean => {
-  // Remove content from selected cells
-  cleanCells(cells);
+  const rng = editor.selection.getRng();
+  const startInTable = SelectorExists.ancestor(SugarElement.fromDom(rng.startContainer), 'table');
+
+  let cellsToClean = cells;
```

Comment:
Instead of the mutation here, this could just be returned via the below function instead. You can simplify the 
"end" slice a little. e.g:
```ts
const cellsToClean = outsideDetails.bind(({ rng, isStartInTable }) => {
    // The endPointCell is the only cell which may have only part of its contents selected.
    const endPointCell = startInTable ? cells[0] : cells[cells.length - 1];
    deleteContentInsideCell(endPointCell, rng, startInTable);
    if (!editor.dom.isEmpty(endPointCell.dom)) {
      return Optional.some(startInTable ? cells.slice(1) : cells.slice(0, -1));
    } else {
      return Optional.none();
    }
}).getOr(cells);
```

Reply:
The `rng` that we get from `outsideDetails` is only the range outside the table, so we can't use it when deleting contents inside a cell, but I can still just grab the Editor range and do it all inside a `bind`